src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -330,15 +316,6 @@ public TemporaryStreamStorage(TemporaryStorageService service, string storageNam
         public long Offset => _memoryMappedInfo!.Offset;
         public long Size => _memoryMappedInfo!.Size;
 
-        public void Dispose()
-        {
-            // Destructors of SafeHandle and FileStream in MemoryMappedFile
-            // will eventually release resources if this Dispose is not called
-            // explicitly
-            _memoryMappedInfo?.Dispose();
-            _memoryMappedInfo = null;
```

Comment:
>_memoryMappedInfo [](http://example.com/codeflow?start=12&length=17)

Does this need to be marked as nullable still?

Reply:
yes.  the underlying TemporaryStreamStorage still operates ina  fashion where that field may be created in the constructor, or created later on.